rosdep/base.yaml

```
@@ -1623,6 +1623,10 @@ libboost-random-dev:
   debian: [libboost-random-dev]
   fedora: [boost-devel]
   ubuntu: [libboost-random-dev]
+libboost-regex-dev:
+  debian: [libboost-regex-dev]
+  fedora: [boost-regex]
```

The `boost-regex` package contains only the runtime library. All boost development files (including headers and linkable libraries) are provided by `boost-devel`. Since the Ubuntu package you're referencing here contains those development files, you should use `boost-devel` here.

```suggestion
  fedora: [boost-devel]
```

---

@cottsay: so there is no way to install only the devel part for a particular Boost library on Fedora?

---

oh my bad I was copying the other boost entries. Thanks!

---

Updated the other boost entries accordingly in https://github.com/ros/rosdistro/pull/23624

---

> so there is no way to install only the devel part for a particular Boost library on Fedora?

Correct - only the runtime libraries are broken out into individual packages, which doesn't typically help us here because we don't typically take a runtime dependency in a package.xml if we take a devel dependency for building...

---

I'm not sure I completely understand your last comment there, but the context in which this PR is submitted: [Generating ‘dev’ and runtime artefacts from ROS packages](https://discourse.ros.org/t/generating-dev-and-runtime-artefacts-from-ros-packages/12448).

---

That exactly describes the behavior I was (poorly) trying to describe above as atypical.

@mikaelarguedas - #23624 looks right at a glance - I can take a closer look in the A.M.

---

@cottsay friendly :bellhop_bell: in case it fell off the plate. Is it possible to get #23624 reviewed/merged?